hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/WriteMarkersFactory.java

```diff
@@ -34,24 +36,28 @@
   private static final Logger LOG = LogManager.getLogger(WriteMarkersFactory.class);
 
   /**
-   * @param markerType the type of markers to use
-   * @param table {@code HoodieTable} instance
+   * @param markerType  the type of markers to use
+   * @param metaClient  {@link HoodieTableMetaClient} instance to use
+   * @param config      Write config
+   * @param context     {@link HoodieEngineContext} instance to use
    * @param instantTime current instant time
-   * @return  {@code WriteMarkers} instance based on the {@code MarkerType}
+   * @return {@code WriteMarkers} instance based on the {@code MarkerType}
    */
-  public static WriteMarkers get(MarkerType markerType, HoodieTable table, String instantTime) {
+  public static WriteMarkers get(
+      MarkerType markerType, HoodieTableMetaClient metaClient, HoodieWriteConfig config,
```

Comment:
can't we get the witeConfig from metaClient itself? trying to see if we can avoid an extra argument.

Reply:
Right.  I revert the changes to the markers-related APIs.